### ISSUE TITLE
Possibility to set external etd logfile?

### ISSUE THREAD

**Issue by varenius** [jive-vlbi/etransfer#17]
etd logs to syslog. So far so good. Except, when you have been away for a long time and want to dig in old etd log output. syslogs rotate, so if one is away long enough the log is gone. Therefore, it could be good to be able to specify an external log directory for etd? If so, it could write logfiles there, perhaps automatically named by start datetime, which could live forever?

**Comment by haavee** [comment]
Please have a look & try using this branch https://github.com/jive-vlbi/etransfer/tree/issue-17

**Comment by varenius** [comment]
Close but no cigar?

```
oper@gyller:/usr/local/src/etransfer/test/etransfer$ git branch
* master
oper@gyller:/usr/local/src/etransfer/test/etransfer$ git checkout issue-17
Branch 'issue-17' set up to track remote branch 'issue-17' from 'origin'.
Switched to a new branch 'issue-17'
oper@gyller:/usr/local/src/etransfer/test/etransfer$ git branch
* issue-17
  master
oper@gyller:/usr/local/src/etransfer/test/etransfer$ ls
argparse11  libudt4hv  libudt5ab  Makefile  README.md  src
oper@gyller:/usr/local/src/etransfer/test/etransfer$ make
[creating version file for etc into Linux-x86_64-native-opt/src/etc_version.cco]
make[1]: Entering directory '/usr/local/src/etransfer/test/etransfer/libudt5ab'
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/md5.o md5.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/common.o common.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/window.o window.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/list.o list.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/buffer.o buffer.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/packet.o packet.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/channel.o channel.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/queue.o queue.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/ccc.o ccc.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/cache.o cache.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/core.o core.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/epoll.o epoll.cpp -c
g++ -fPIC -Wall -Wextra -O2 -DLINUX -finline-functions -fno-strict-aliasing -DAMD64 -o ../Linux-x86_64-native-opt/libudt5ab/api.o api.cpp -c
ar -rcs ../Linux-x86_64-native-opt/libudt5ab/libudt5ab.a ../Linux-x86_64-native-opt/libudt5ab/md5.o ../Linux-x86_64-native-opt/libudt5ab/common.o ../Linux-x86_64-native-opt/libudt5ab/window.o ../Linux-x86_64-native-opt/libudt5ab/list.o ../Linux-x86_64-native-opt/libudt5ab/buffer.o ../Linux-x86_64-native-opt/libudt5ab/packet.o ../Linux-x86_64-native-opt/libudt5ab/channel.o ../Linux-x86_64-native-opt/libudt5ab/queue.o ../Linux-x86_64-native-opt/libudt5ab/ccc.o ../Linux-x86_64-native-opt/libudt5ab/cache.o ../Linux-x86_64-native-opt/libudt5ab/core.o ../Linux-x86_64-native-opt/libudt5ab/epoll.o ../Linux-x86_64-native-opt/libudt5ab/api.o
make[1]: Leaving directory '/usr/local/src/etransfer/test/etransfer/libudt5ab'
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etc.cco src/etc.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/reentrant.cco src/reentrant.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etdc_fd.cco src/etdc_fd.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etdc_etdserver.cco src/etdc_etdserver.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etdc_debug.cco src/etdc_debug.cc
In file included from src/etdc_debug.cc:20:0:
/usr/local/src/etransfer/test/etransfer/src/etdc_debug.h: In constructor ‘etdc::detail::filelog_streambuf<Props>::filelog_streambuf(const string&, const string&)’:
/usr/local/src/etransfer/test/etransfer/src/etdc_debug.h:277:83: error: ‘::strerror’ has not been declared
           e << "Failed to open logfile: " << __m_fName << ", " << ::strerror(errno)
                                                                     ^~~~~~~~
/usr/local/src/etransfer/test/etransfer/src/etdc_debug.h:277:83: note: suggested alternative: ‘perror’
           e << "Failed to open logfile: " << __m_fName << ", " << ::strerror(errno)
                                                                     ^~~~~~~~
                                                                                   perror
/usr/local/src/etransfer/test/etransfer/src/etdc_debug.h:284:64: error: ‘::strerror’ has not been declared
                         e << "Failed to replace stderr: " << ::strerror(errno)
                                                                ^~~~~~~~
/usr/local/src/etransfer/test/etransfer/src/etdc_debug.h:284:64: note: suggested alternative: ‘perror’
                         e << "Failed to replace stderr: " << ::strerror(errno)
                                                                ^~~~~~~~
                                                                perror
Makefile:200: recipe for target 'Linux-x86_64-native-opt/src/etdc_debug.cco' failed
make: *** [Linux-x86_64-native-opt/src/etdc_debug.cco] Error 1
```

**Comment by haavee** [comment]
I've updated issue-17 branch, a `git pull` and re-make should fix this

**Comment by varenius** [comment]
```
oper@gyller:/usr/local/src/etransfer/test/etransfer$ make
[creating version file for etc into Linux-x86_64-native-opt/src/etc_version.cco]
make[1]: Entering directory '/usr/local/src/etransfer/test/etransfer/libudt5ab'
make[1]: Leaving directory '/usr/local/src/etransfer/test/etransfer/libudt5ab'
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etc.cco src/etc.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/reentrant.cco src/reentrant.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etdc_fd.cco src/etdc_fd.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etdc_etdserver.cco src/etdc_etdserver.cc
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etdc_debug.cco src/etdc_debug.cc
g++  -o Linux-x86_64-native-opt/etc  Linux-x86_64-native-opt/src/etc.cco  Linux-x86_64-native-opt/src/reentrant.cco  Linux-x86_64-native-opt/src/etdc_fd.cco  Linux-x86_64-native-opt/src/etdc_etdserver.cco  Linux-x86_64-native-opt/src/etdc_debug.cco Linux-x86_64-native-opt/src/etc_version.cco  -lm -lnsl -lrt -ldl -L./Linux-x86_64-native-opt/libudt5ab -ludt5ab -lpthread 
[creating version file for etd into Linux-x86_64-native-opt/src/etd_version.cco]
g++ -fPIC -O2 -Wall -W -Werror -Wextra -pedantic -DB2B=native -D_POSIX_C_SOURCE=200809L -D_XOPEN_SOURCE=700 -D_GNU_SOURCE -U_GNU_SOURCE -D__STDC_FORMAT_MACROS -Wcast-qual -Wwrite-strings -Wredundant-decls -Wfloat-equal -Wshadow -D_FILE_OFFSET_BITS=64 -pthread -D_REENTRANT -D_POSIX_PTHREAD_SEMANTICS  -std=c++11  -I/usr/local/src/etransfer/test/etransfer/src -I/usr/local/src/etransfer/test/etransfer/argparse11 -I/usr/local/src/etransfer/test/etransfer/libudt5ab -c -o Linux-x86_64-native-opt/src/etd.cco src/etd.cc
In file included from /usr/local/src/etransfer/test/etransfer/src/etdc_fd.h:29:0,
                 from src/etd.cc:21:
/usr/local/src/etransfer/test/etransfer/src/etdc_debug.h: In member function ‘int etdc::detail::filelog_streambuf<Props>::sync() [with Props = {char, std::char_traits<char>}]’:
/usr/local/src/etransfer/test/etransfer/src/etdc_debug.h:297:32: error: ignoring return value of ‘ssize_t write(int, const void*, size_t)’, declared with attribute warn_unused_result [-Werror=unused-result]
                         ::write( 2, __m_buf.c_str(), __m_buf.size() );
                         ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
Makefile:200: recipe for target 'Linux-x86_64-native-opt/src/etd.cco' failed
make: *** [Linux-x86_64-native-opt/src/etd.cco] Error 1
```

**Comment by haavee** [comment]
Ur compiler is a bit stricter than mine ;-)
Try again? (`git pull` && `make`)

**Comment by varenius** [comment]
Seems to work fine now, great! Only one question; the logfile name is created containg spaces, e.g. 'etd 2021-09-11 05:35:35.69'. Is this kosher, or should the spaces maybe instead be e.g. underscores?

**Comment by haavee** [comment]
It can be anything - this was the easiest because I reused existing code that generates the time stamps for the actual logmessages themselves, so the file naming would be consistent by design. It would be a minor modification to change them into underscores.

**Comment by varenius** [comment]
Personally I don't have any opinion. But paths without spaces generally are a little more friendly, since one does not have to escape them traversing trees with cd etc. It's really a minor thing :). Anyway, the desired feature seems to work fine! Feel free to alter to non-spaces if you want (e.g. replacing with underscores) and then merge this into main.

**Comment by haavee** [comment]
I notice in my shell that the `:`'s - as in the actual time part - are also being escaped:
```bash
> less /tmp/etd-hv/etd_2021-09-14_11\:12\:33.92
```
Is that an issue too?

**Comment by varenius** [comment]
Good question. Probably yes. Might as well use underscore also for the colons? It's straightforward to parse anyway.

**Comment by haavee** [comment]
I guess "XXhYYmZZ.ZZs" might be even better - if the `\:` go to `_` it's just a "random" sequence of numbers

**Comment by varenius** [comment]
Right. You could then even have the "T" between the day and the hour, to make it almost iso 8601 :). So underscore just after etd?

**Comment by varenius** [comment]
After some iterations via mattermost, it seems the proposed branch works fine. Perhaps worth merging into main? I'd appreciate this capability for my everyday etd usage.

**Comment by haavee** [comment]
Merged to `master`, this closes issue #17
